src/block/Cactus.php

```diff
@@ -61,6 +62,21 @@ public function onEntityInside(Entity $entity) : bool{
 		return true;
 	}
 
+	public function onNearbyBlockChange2(int $flags) : void{
+		if(($flags & NearbyBlockChangeFlags::DOWN) !== 0){
+			$supportBlock = $this->getSide(Facing::DOWN);
+			if(!$supportBlock->hasSameTypeId($this) && !$supportBlock->hasTypeTag(BlockTypeTags::SAND)){
+				$this->position->getWorld()->useBreakOn($this->position);
```

Comment:
Could be set a new `break` var to avoid entering in the loop later

Reply:
Maybe. This whole self-destruct thing needs some cleanup work tbh, so I wouldn't think too much about it.